src/Compilers/Core/Portable/Symbols/Nullability.cs

```diff
@@ -0,0 +1,117 @@
+    /// </code>
+    /// The use of <code>o1</code> is inferred to be <see cref="NonNull"/>, despite 
+    /// being declared <see cref="MaybeNull"/>, and no warning is reported when it 
+    /// is dereferenced. However, the first dereference of <code>o2</code> has not 
+    /// been inferred to be <see cref="NonNull"/>, so a warning is reported. The second 
+    /// dereference occurs inside an <code>if</code> statement check for <see langword="null"/>,
+    /// so the compiler can infer that <code>o2</code> must not be <see langword="null"/> 
+    /// inside the <code>if</code> statement.
+    /// </para>
+    /// <para>
+    /// The states of value types and nullable value types are not dynamically tracked by
+    /// the compiler, so they will always be their initial state. Value types will always
+    /// have a nullability of <see cref="NonNull"/>, and nullable value types will always
+    /// have a nullability of <see cref="MaybeNull"/>.
+    /// </para>
+    /// <para>
+    /// Unconstrained type parameters are a unique superposition of both <see cref="NonNull"/>
+    /// and <see cref="MaybeNull"/>. They must be checked for <see langword="null"/> before
+    /// being dereferenced, but are treated as <see cref="NonNull"/> for the purposes of
+    /// assignment and return type warnings. As an example:
+    /// <code>
+    /// T M&lt;T&gt;(T t1)
+    /// {
+    ///     t1.ToString(); // Warning reported for dereferencing t1
+    ///     t1?.ToString(); // No warning reported for using ?. operator
+    ///     T t2 = default(T2); // Warning reported for assigning possibly null into t2
+    ///     return t2; // Warning reported for returning possibly null
+    /// }
+    /// </code>
+    /// This is because both <code>string</code> and <code>string?</code> can be substituted
+    /// for <code>T</code>, so the user must account for the strictest possible scenario
+    /// in all cases. For this reason, variables and values of type unconstrained type
+    /// parameter are considered <see cref="MaybeNull"/> until checked.
+    /// </para>
+    /// </remarks>
+    public enum Nullability
+    {
+        /// <summary>
+        /// The given <see cref="SyntaxNode"/> or <see cref="IOperation"/> does not represent 
+        /// a variable or value, and does not have a nullability.
+        /// </summary>
+        /// <remarks>
+        /// This is used for statements, and for expressions that do not have nullabilities,
+        /// such as method groups.
+        /// </remarks>
+        NotApplicable = 0,
+        /// <summary>
+        /// There is no information on the current nullable state of the variable
+        /// or value.
+        /// </summary>
+        /// <remarks>
+        /// This is used for legacy code scenarios, where a legacy API that does
+        /// not provide nullability information is being used. Expressions that
+        /// have unknown nullability are also referred to as oblivious expressions,
```

Comment:
is there a good reason to have both names?  if we refer to them as Oblivious, shoudl the name of this enum member be that?